CONTRIBUTING.md

```diff
@@ -31,6 +31,42 @@ The Dart source code in this repo follows the:
 
 You should familiarize yourself with those guidelines.
 
+## Pubspec versioning
+
+The first change to a package after a publish needs to increment the package's
+pubspec version. Additional changes may also require pubspec version updates;
+see our
+[package maintenance wiki](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change)
+for a description of the process.
+
+## Changelog entries
+
+Most changes should be accompanied by a new entry in the CHANGELOG.md file. A
+good rule of thumb is that if the change is user-facing, it should have a
+changelog entry. This includes changes to documentation - like a README.md file
+update - or changes to dartdoc documentation.
+
+Some changes which are exempt are:
+
+- infrastructure changes (changes inside the `.github` directory)
+- changes to tests
+
+And generally, the pubspec version and the changelog version should always
+agree.
+
+## Testing
+
+Changes to dart-lang repos must have tests, except for PRs that:
+
+- only affect comments (including documentation)
```

Comment:
Maybe rather say something along the lines of 
```
Changes to dart-lang repos which modify production code must have tests, except for PRs that 
- are generated by automated bots (rollers)
- have an explicit exemption from a repository contributor
```
? But this is probably bikeshedding.

Reply:
Hmm, I think I'd like to keep this list as-is (it's borrowed from the flutter PR template, and I think is pretty reasonable). But happy to update this section on an on-going basis.

I did adjust the final sentence to 'Repository contributors may indicate that a PR should have a test, even for the general exemptions above.'.